src/python/WMComponent/JobSubmitter/JobSubmitterPoller.py

```diff
@@ -754,7 +754,16 @@ def submitJobs(self, jobsToSubmit):
         myThread.transaction.begin()
 
         # Run the actual underlying submit code using bossAir
-        successList, failList = self.bossAir.submit(jobs=jobList)
+        try:
+            successList, failList = self.bossAir.submit(jobs=jobList)
+ 
+        except Exception as ex:
+            msg = "Error submitting jobs: %s" % str(ex)
+            logging.error(msg)
+            myThread.logdbClient.post("JobSubmitter_submitWork", msg, "error")
+            myThread.transaction.rollback()
+            raise WMException(msg) from ex
```

Comment:
i am not sure i understand the need for this rollback, since it seems that if you raise a wmexception that the rollback already happens here https://github.com/dmwm/WMCore/blob/beefc74a275745e8191ea922e3ee482f73f76cbd/src/python/WMComponent/JobSubmitter/JobSubmitterPoller.py#L833

Reply:
Hi @mapellidario, I agree it was redundant, I have received the rollback from the submitJobs method.